packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -783,6 +750,50 @@ class CupertinoTextField extends StatefulWidget {
   /// {@macro flutter.services.TextInputConfiguration.enableIMEPersonalizedLearning}
   final bool enableIMEPersonalizedLearning;
 
+  /// {@macro flutter.widgets.EditableText.contextMenuBuilder}
+  ///
+  /// If not provided, will build a default menu based on the platform.
+  ///
+  /// See also:
+  ///
+  ///  * [DefaultCupertinoTextSelectionToolbar], which is built by default.
+  final EditableTextToolbarBuilder? contextMenuBuilder;
+
+  /// The platform that the current `defaultTargetPlatform` will be treated as.
+  ///
+  /// The Cupertino library has no access to Material, so it cannot build things
+  /// like the Material text selection toolbars on the platforms that use them.
+  /// Instead it will treat those non-Cupertino platforms as if they are the
+  /// platform returned here.
+  ///
+  /// For example, using a CupertionTextField on an Android device will show the
+  /// iOS text selection toolbar.
+  static TargetPlatform get _cupertinoPlatform {
+    switch (defaultTargetPlatform) {
+      case TargetPlatform.android:
+      case TargetPlatform.iOS:
+      case TargetPlatform.fuchsia:
+        return TargetPlatform.iOS;
+      case TargetPlatform.linux:
+      case TargetPlatform.windows:
+      case TargetPlatform.macOS:
+        return TargetPlatform.macOS;
+    }
+  }
+
+  static Widget _defaultBuildContextMenu(BuildContext context, EditableTextState editableTextState, Offset primaryAnchor, [Offset? secondaryAnchor]) {
```

Comment:
It feels like this is duplicating code between TextFields?

Reply:
This will be much simpler once the deprecations go away, I think it's the toolbarOptions stuff that seems so verbose here.